go.mod

```diff
@@ -1,20 +1,23 @@
 module github.com/bitcoin-sv/go-broadcast-client
 
-go 1.20
+go 1.21.3
+
+toolchain go1.21.5
```

Comment:
can we have those versions aligned - I mean both at the same version 1.21.5

Reply:
Yes, we can. The question is, should we?

From version go1.21, the `go line` represents the **minimal** go version that satisfies dependencies versions. This line is also the minimum required go version to use this lib in other apps.

The `toolchain line` is the **preferred** version of go to use when running this project and is chosen in the specific way, described in this document (https://go.dev/doc/toolchain) - screenshot below.

![image](https://github.com/bitcoin-sv/go-broadcast-client/assets/127198012/387ab8f9-5c8b-4b6f-9412-e62f441f5a41)

In order to have these versions aligned, we can update the `go` version to match the `toolchain` version (if the versions are aligned, the toolchain line will disappear), or specify a custom, lower toolchain version. Updating the go version in a lib would force all users of the lib to update their projects to the highest version, so maybe it's not the best choice. Lowering the toolchain version is also not perfect, because toolchain selection is automatic and every time we will run `go get -u ./...` or something similar, it will choose this toolchain again. I would leave it as it is.

WDYT?

Reply:
I updated go to the newest version. Now they are aligned so toolchain line was automatically removed :)